File: Programmers/Level 2/10.py
def solution(n):
    answer = []
    number = ['1','2','4']
    while n != 0:
        if n % 3 == 1:
            answer.append('1')
            n //= 3
        elif n % 3 == 2:
            answer.append('2')
            n //= 3
        else:
            answer.append('4')
            n = (n//3) - 1
    answer.reverse()
    answer = ''.join(answer)
    return answer

File: Programmers/Level 2/10.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(int n) {
    string answer = "";
    int arr[3] = { 4,1,2 };
    //arr[0] = 4
    //arr[1] = 1
    //arr[2] = 2
    
    while (n != 0) {
        answer += to_string(arr[n % 3]);
        if (n % 3 == 0)
            n = (n/3) - 1;
        else
            n /= 3;
    }

    reverse(answer.begin(), answer.end());
    return answer;
}
